metaflow/task.py

```diff
@@ -709,24 +709,29 @@ def run_step(
                         name="end",
                         payload={**task_payload, "msg": "Task ended"},
                     )
-
-                attempt_ok = str(bool(self.flow._task_ok))
-                self.metadata.register_metadata(
-                    run_id,
-                    step_name,
-                    task_id,
-                    [
-                        MetaDatum(
-                            field="attempt_ok",
-                            value=attempt_ok,
-                            type="internal_attempt_status",
-                            tags=["attempt_id:{0}".format(retry_count)],
-                        )
-                    ],
-                )
+                try:
+                    # persisting might fail due to unpicklable artifacts.
+                    output.persist(self.flow)
+                except Exception as ex:
+                    self.flow._task_ok = False
```

Comment:
Let's add `self.flow._exception = MetaflowExceptionWrapper(ex)` here as well. Maybe a print statement as well.

Reply:
what is the retained exception used for internally? I'm also not sure if adding a print will help in this case.

I'm re-raising the exception here, but I've noticed that the stderr at this stage is not captured by task logs anymore. Maybe that is by choice, but it leads to the PR being a halfway fix for the issue where a non-zero exit task is correctly marked as failed, but there are no logs to determine why it failed.